packages/pwa/app/utils/site-utils.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {getConfig} from 'pwa-kit-react-sdk/ssr/universal/utils'
+import {getParamsFromPath} from './utils'
+
+/**
+ * This functions takes an url and returns a site object,
+ * an error will be thrown if no url is passed in or no site is found
+ * @param {string} url
+ * @returns {object} site - a site object
+ */
+export const resolveSiteFromUrl = (url) => {
+    if (!url) {
+        throw new Error('url is required to find a site object.')
+    }
+    const {pathname, search} = new URL(url)
```

Comment:
I think at this point, we can also call `pathToUrl` to make sure the given url is the absolute url.

I saw how other files are calling `resolveSiteFromUrl`. They typically would call `pathToUrl` first. So I think here there's an opportunity to have 1 less step for the callers.

Reply:
`pathToUrl` will return a string, while the reason we call new URL here is to be able to grab the pathname and search easily without us performing string split

Reply:
@alexvuong I mean, how can we save a step for other developers? Given the latest code, I'm thinking to call `absoluteUrl` like this.

```suggestion
    const {pathname, search} = new URL(absoluteUrl(url))
```

So when others call `resolveSiteFromUrl`:
- no need to do this `resolveSiteFromUrl(absoluteUrl(...))`
- they can simply call `resolveSiteFromUrl(url)`

Reply:
gotcha!